saleor/checkout/tests/test_calculations.py

```diff
@@ -1156,3 +1177,93 @@ def test_fetch_checkout_with_prior_price_none(
     line.refresh_from_db()
     assert line.prior_unit_price_amount is None
     assert line.currency is not None
+
+
+def test_calculate_checkout_total_with_gift_card(
+    checkout_with_item_and_gift_card,  # fixture that includes a gift card applied
+):
+    # given
+    checkout = checkout_with_item_and_gift_card
+    manager = get_plugins_manager(allow_replica=False)
+    lines_info, _ = fetch_checkout_lines(checkout)
+    checkout_info = fetch_checkout_info(checkout, lines_info, manager)
+
+    # Add a gift card balance greater than total
+    checkout.get_total_gift_cards_balance = lambda db_name: TaxedMoney(
```

Comment:
We should create a variable only if it is needed.  IMO, it's a good case for using Mock instead of lambda. 
```suggestion
    checkout.get_total_gift_cards_balance = lambda _: TaxedMoney(
```